components/esp_modem/src/esp_modem_dte.cpp

```diff
@@ -69,6 +69,23 @@ command_result DTE::command(const std::string &cmd, got_line_cb got_line, uint32
     return command(cmd, got_line, time_ms, '\n');
 }
 
+bool DTE::exit_cmux()
+{
+    auto cmux_term = static_cast<CMuxInstance*>(term.get())->get_cmux();
+    auto ejected = cmux_term->deinit_and_eject();
+    if (ejected == std::tuple(nullptr, nullptr, 0)) {
+        return false;
+    }
+    // deinit succeeded -> swap the internal terminals with those ejected from cmux
+    auto term_orig = std::move(term);
+    auto other_term_orig = std::move(other_term);
+    term = std::move(std::get<0>(ejected));
+    buffer = std::move(std::get<1>(ejected));
+    buffer_size = std::get<2>(ejected);
+    command_term = term.get(); // use command terminal as previously
```

Comment:
About the lifetime control here. 
We have on `CMUX` usage of `std::unique_ptr` and here we gave up the automatic control to have a raw pointer.

Reply:
I see, this is a design flaw introduced by the CMUX mode. I could make it a `shared_ptr<>`, but that would mean promote every `unique_ptr<>` to the shared pointer, which I was a bit reluctant. I'll see if I can fix it...

Reply:
Refactored to used `shared_ptr<>` for both types of terminals, so they could be attached to `CMux` when switching to CMUX mode and detached when switching back.